core/domain/topic_services.py

```diff
@@ -255,15 +258,29 @@ def apply_change_list(topic_id, change_list):
             topic.
 
     Returns:
-        Topic. The resulting topic domain object.
+        dict, list(int), list(int). A dict with two fields: topic and
+            subtopic_pages containing the updated domain objects of each one,
+            a list of ids of the deleted subtopics and a list of ids of the
+            newly created subtopics.
     """
     topic = get_topic_by_id(topic_id)
+    subtopic_pages = subtopic_page_services.get_all_subtopic_pages_in_topic(
+        topic_id)
+    newly_created_subtopic_ids = []
```

Comment:
Perhaps better to have newly_created_subtopics here, so you can edit them.

Reply:
Do you mean newly_created_subtopic_pages? as the subtopics would be in topic.subtopics, right?
Even then, subtopic_pages would have all of these.

Reply:
First question: yep, that's right!

Second question: depends on how you're handling deletions. May be hard to figure out whether something needs to actually be deleted (vs not being created at all) if you bundle them together.